stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/PrintAny.java

```diff
@@ -48,7 +51,8 @@ public static void print(Strand strand, ArrayValue values) {
         for (int i = 0; i < values.size(); i++) {
             value = values.get(i);
             if (value != null) {
-                out.print(value.toString());
+                BType type = TypeChecker.getType(value);
```

Comment:
@irshadnilam Shall we do the refactoring you suggested to StringUtils.getStringValue?
resolving type can be . moved inside the getStringValue itself.

Reply:
+1 will refactor it.

Reply:
@gimantha https://github.com/ballerina-platform/ballerina-lang/pull/18625/commits/d889c75853cce78ce0c958823fad58c191226ac3 refactors it